File: blockchain/src/test/java/pt/um/lei/masb/blockchain/test/TestBlockchain.java
package pt.um.lei.masb.blockchain.test;

import org.junit.jupiter.api.Test;
import pt.um.lei.masb.blockchain.BlockChain;

import javax.persistence.spi.PersistenceProviderResolverHolder;
import java.util.Objects;

import static org.junit.jupiter.api.Assertions.*;

class TestBlockchain {
    @Test
    void testBlockchainInit() {
        try {
            var resolver = PersistenceProviderResolverHolder.getPersistenceProviderResolver();
            var providers = resolver.getPersistenceProviders();


            //There is a persistance provider.
            assertTrue(providers.stream().anyMatch(Objects::nonNull));
            providers.forEach(System.out::println);
            //providers.forEach(pr -> System.out.println(pr.getClass().getClassLoader().getResource("META-INF/persistence.xml")));
            //providers.stream().findAny().ifPresent(t -> assertNotNull(t.createEntityManagerFactory(null, null)));
            var bl = BlockChain.getInstance();
            var b = bl.newBlock();
            var c = b.getCoinbase();
            assertNotNull(c);
        } catch (Exception e) {
            fail(e);
        }
    }
}

File: blockchain/src/test/kotlin/pt/um/lei/masb/test/TestBlockchain.kt
package pt.um.lei.masb.test

import org.junit.jupiter.api.Test

class TestBlockchain {
    @Test
    fun testBlockchainInit() {
/*        try {
            val resolver = PersistenceProviderResolverHolder.getPersistenceProviderResolver()
            val providers = resolver.persistenceProviders


            //There is a persistance provider.
            assertTrue(providers.stream().anyMatch(Objects::nonNull))
            providers.forEach(System.out::println)

            //providers.forEach(pr -> System.out.println(pr.getClass().getClassLoader().getResource("META-INF/persistence.xml")));
            //providers.stream().findAny().ifPresent(t -> assertNotNull(t.createEntityManagerFactory(null, null)));

            val bl = BlockChain()
            val s = bl.registerSideChainOf(TemperatureData::class.java).getSideChainOf(TemperatureData::class.java)
            val b = s.newBlock()
            val c = b.coinbase

            assertNotNull(c)
        } catch (e : Exception) {
            fail<Nothing>(e)
        }
*/
    }
}
